client-js/app/elements/app-link.ts

```diff
@@ -1,8 +1,11 @@
+import {PolymerElement} from "@polymer/polymer";
+import {customElement, property} from "@polymer/decorators";
+
 const MOUSE_MAIN_BUTTON = 0;
 
-@component('app-link')
-@extend('a')
-export class AppLink extends polymer.Base {
+@customElement('app-link')
+// @extend('a')  // TODO: https://polymer-library.polymer-project.org/3.0/docs/devguide/custom-elements#extending-elements
+export class AppLink extends PolymerElement {
```

Comment:
The fix for this might be to `extend HTMLLinkElement` instead of `PolymerElement`. However: "Polymer does not currently support extending built-in elements." (https://polymer-library.polymer-project.org/3.0/docs/devguide/custom-elements#extending-elements)

Reply:
Yeah, we'll have to just try to match as much of the behavior of the built-in `<a>` element as possible, since we can't actually extend `<a>`.